en_US/changes/changes-ce-v5.md

```diff
@@ -1,5 +1,212 @@
+  This change also added validation for the input date format.
+
+- [#12672](https://github.com/emqx/emqx/pull/12672) Changed the process for generating the node boot configuration by incorporating the loading of `{data_dir}/configs/cluster.hocon`. Previously, changes to logging configurations made via the Dashboard and saved in `{data_dir}/configs/cluster.hocon` were only applied after the initial boot configuration was generated using `etc/emqx.conf`, leading to potential loss of some log segment files due to late reconfiguration.
+
+  Now, both `{data_dir}/configs/cluster.hocon` and `etc/emqx.conf` are loaded concurrently, with settings from `emqx.conf` taking precedence, to create the boot configuration.
+
+- [#12696](https://github.com/emqx/emqx/pull/12696) Fixed an issue where attempting to reconnect an action or source could lead to wrong error messages being returned in the HTTP API.
+
+- [#12714](https://github.com/emqx/emqx/pull/12714) Fixed inaccuracies in several metrics reported by the `/prometheus/stats` endpoint of the Prometheus API. The correction applies to the following metrics:
+
+  - `emqx_cluster_sessions_count`
+  - `emqx_cluster_sessions_max`
+  - `emqx_cluster_nodes_running`
+  - `emqx_cluster_nodes_stopped`
+  - `emqx_subscriptions_shared_count`
+  - `emqx_subscriptions_shared_max`
+
+  Additionally, this fix rectified an issue within the `/stats` endpoint concerning the `subscriptions.shared.count` and `subscriptions.shared.max` fields. Previously, these values failed to update promptly following a client's disconnection or unsubscription from a Shared-Subscription.
+
+- [#12715](https://github.com/emqx/emqx/pull/12715) Fixed a crash that could occur during configuration updates if the connector for the ingress data integration source had active channels.
+
+- [#12740](https://github.com/emqx/emqx/pull/12740) Fixed an issue when durable sessions could not be kicked out.
+
+- [#12768](https://github.com/emqx/emqx/pull/12768) Addressed a startup failure issue in EMQX version 5.4.0 and later, particularly noted during rolling upgrades from versions before 5.4.0. The issue was related to the initialization of the routing schema when both v1 and v2 routing tables were empty.
+
+  The node now attempts to retrieve the routing schema version in use across the cluster instead of using the v2 routing table by default when local routing tables are found empty at startup. This approach mitigates potential conflicts and reduces the chances of diverging routing storage schemas among cluster nodes, especially in a mixed-version cluster scenario. 
+
+  If conflict is detected in a running cluster, EMQX writes instructions on how to manually resolve it in the log as part of the error message with `critical` severity. The same error message and instructions will also be written in the `error` level log to make sure this message will not get lost even if no log handler is configured.

```

Comment:
> in the `error` level log

this is incorrect, "standard error" is not `error` level log, it is standard linux data stream that can be used by programs to write error messages. See, for example, https://www.howtogeek.com/435903/what-are-stdin-stdout-and-stderr-on-linux/.

Standard error data stream is always available to programs, that's why we write error message and instructions there. Using `error` level log would make no sense, since `critical` is of higher severity, and each error message with `critical` severity will end up in `error` level log anyway. We try to handle situation when user disabled all standard log handlers in the application. In this case they will see the message anyway.

Reply:
Sorry for the mistakes and thank you for the clear explanation. I have corrected the descriptions.